app/Helpers/filesystem.php

```diff
@@ -211,3 +211,39 @@ function getFileNameExtension($fileName): ?string
         return $explodedFileName[1] ?? null;
     }
 }
+
+if (!function_exists('arraysHaveCommonKey')) {
+    /**
+     * Checks if 2 arrays share common keys.
+     *
+     * @param array $array1
+     * @param array $array2
+     * @param bool $returnKeys
+     *
+     * @return bool|array
+     */
+    function arraysHaveCommonKey(array $array1, array $array2, bool $returnKeys = false): bool | array
```

Comment:
Comparing if two arrays have commn keys, it can be done using "array_intersect_keys()".